osu.Game/Overlays/Comments/CommentsContainer.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Framework.Graphics;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+using osu.Game.Online.API.Requests.Responses;
+using System.Threading;
+
+namespace osu.Game.Overlays.Comments
+{
+    public class CommentsContainer : CompositeDrawable
+    {
+        private readonly CommentableType type;
+        private readonly long id;
+
+        public readonly Bindable<CommentsSortCriteria> Sort = new Bindable<CommentsSortCriteria>();
+        public readonly BindableBool ShowDeleted = new BindableBool();
+
+        [Resolved]
+        private IAPIProvider api { get; set; }
+
+        [Resolved]
+        private OsuColour colours { get; set; }
+
+        private GetCommentsRequest request;
+        private CancellationTokenSource loadCancellation;
+        private int currentPage;
+        private int loadedTopLevelComments;
```

Comment:
I'm not sure the performance benefit of having this in a field outweighs the cost of potential bugs when the state of this gets polluted.

Reply:
Are you talking about `loadedTopLevelComments` line?

Reply:
Yeah, I'd honestly ideally want to get rid of that field, it seems like it'll cause problems down the line. Would rather iterate over the comments every time and be sure the count is correct than risk potential bugs arising from forgetting to update it.

Reply:
I'd also like to see it removed.